src/core/varinfo.jl

```diff
@@ -169,56 +171,50 @@ end
 
 # Main behaviour control of rand() depending on sampler type and if sampler inside
 rand(vi::VarInfo, vn::VarName, dist::Distribution, spl::Sampler, inside=true) = begin
```

Comment:
Yes we can. But PG still need to test the association of a variable and a sampler in order to run `produce(1.0)` in `assume()`.